### PR TITLE
Feat/add survey imgs

### PR DESCRIPTION
**Pull request by emontealvo**
# Pull request checklist

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## Current behavior

- No images on style survey
- No style survey submission handler

## New behavior

- Style survey is now populated with images 
- On submission, a string is created to send to the back-end

## Other information

image
#### This is the string format that's sent to back end
image

### CODE REVIEW

File: code/web/src/modules/styleSurvey/StyleSurvey.js [hunk 1]
```diff
@@ -1,13 +1,22 @@
 // Imports
 import React, { PureComponent } from 'react'
 import PropTypes from 'prop-types'
+import SurveyChoices from './surveyChoices'
+import accessories from '../../../public/images/surveyImages/accessories/index'
+import bottoms from '../../../public/images/surveyImages/bottoms/index'
+import dresses from '../../../public/images/surveyImages/dresses/index'
+import shoes from '../../../public/images/surveyImages/shoes/index'
+import tops from '../../../public/images/surveyImages/tops/index'
```
Comment (emontealvo) [inline code/web/src/modules/styleSurvey/StyleSurvey.js:9]: Import survey content

File: code/web/src/modules/styleSurvey/StyleSurvey.js [hunk 2]
```diff
@@ -56,33 +62,36 @@ class StyleSurvey extends PureComponent {
...
-          choices={shoes}
-          handleChange={this.handleChange}
-          choiceCategory='shoes'
-        />
-        <SurveyChoices
-          choices={tops}
-          handleChange={this.handleChange}
-          choiceCategory='tops'
-        />
+      <section>
```
Comment (jeremiahblackol) [inline code/web/src/modules/styleSurvey/StyleSurvey.js:74]: nice and dry

Comment (perryr16) [reply code/web/src/modules/styleSurvey/StyleSurvey.js:74]: so dry

File: code/web/src/modules/styleSurvey/StyleSurvey.js [hunk 3]
```diff
@@ -26,65 +35,73 @@ class StyleSurvey extends PureComponent {
...
+      id: choice.id
+    }
+    console.log(status)
+    if (status) {
+      this.setState({
+        [category]: this.state[category].filter(previousChoice => previousChoice.id !== choice.id)
+      })
+    } else {
+      this.setState({
+        [category]: [...this.state[category], selection]
```
Comment (perryr16) [inline code/web/src/modules/styleSurvey/StyleSurvey.js:52]: I take it this is the creation of the style string

File: code/web/src/modules/styleSurvey/StyleSurvey.js [hunk 4]
```diff
@@ -26,65 +35,73 @@ class StyleSurvey extends PureComponent {
     // this.props.someAction()
   }
 
-  handleChange(event) {
-    // event.preventDefault()
-    // 
-    this.setState({[event.target.name]: [...this.state[event.target.name], event.target.name]})
+  handleChoice = (choice, status) => {
```
Comment (emontealvo) [inline code/web/src/modules/styleSurvey/StyleSurvey.js:38]: This adds/remove a survey choice depending on whether is checked or not

File: code/web/src/modules/styleSurvey/StyleSurvey.js [hunk 5]
```diff
@@ -26,65 +35,73 @@ class StyleSurvey extends PureComponent {
...
+      })
+    }
   }
 
-  generateGarmentCheckboxes = (/*array of garments*/) => {
-    // create a div or section to hold the garments in the section
-    // iterate over array of garments
-    // for each garment return a check box with a key or id, relative to its style
-    // will do one for each garment category
+  handleSubmission = (event) => {
```
Comment (emontealvo) [inline code/web/src/modules/styleSurvey/StyleSurvey.js:57]: On submission, this create a style string to send to the back end

File: code/web/src/modules/styleSurvey/surveyChoices.js [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+      margin: '5px',
+      padding: '3px',
+      border: '3px double #333'
+    }
+    const unselectedStyle = {
+      margin: '5px',
+      padding: '3px'
+    }
+    return selectionStatus ? selectedStyle : unselectedStyle
+  }
```
Comment (emontealvo) [inline code/web/src/modules/styleSurvey/surveyChoices.js:30]: An attempt to create dynamic styles, not quite finished.

File: code/web/src/modules/styleSurvey/surveyChoices.js [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+  const handleChange = (event) => {
+    const choice = event.target.closest('span')
+    const selected = choice.getAttribute('aria-checked') === 'true'
+    handleChoice(choice, selected)
+    if (selected) {
+      choice.setAttribute('aria-checked', 'false')
+    } else {
+      choice.setAttribute('aria-checked', 'true')
+    }
+  }
```
Comment (emontealvo) [inline code/web/src/modules/styleSurvey/surveyChoices.js:15]: aria-checked is a helpful accessibility attribute that helps keep track of what's selected.

File: code/web/src/modules/styleSurvey/surveyChoices.js [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+          <img style={{
+            height: '200px',
+            width: '150px',
+            display: 'inline'
+          }}
+            src={`${ APP_URL }/images/surveyImages/${category}/${choice.image}`}
+          />
+        </span>
+    )
+    })
```
Comment (emontealvo) [inline code/web/src/modules/styleSurvey/surveyChoices.js:55]: Bringing in images from a front end folder proved tricky. It also took some creativity to recreate a checkbox with out having any checkbox inputs to disrupt content flow.

File: code/web/webpack.config.babel.js [hunk 9]
```diff
@@ -20,6 +20,13 @@ const config = {
           loader: 'babel-loader',
         },
         exclude: /node_modules/
+      }, {
+        test: /\.(png|jpe?g|gif)$/i,
+        use: [
+          {
+            loader: 'file-loader',
+          },
```
Comment (emontealvo) [inline code/web/webpack.config.babel.js:28]: Webpack loader to handle image files

**Pull request merged**